modules/core/include/opencv2/core/base.hpp

```diff
@@ -278,6 +278,15 @@ enum BorderTypes {
     BORDER_ISOLATED    = 16 //!< do not look outside of ROI
 };
 
+//! Flags for nanMask() function
+enum NanMaskFlags {
+    MASK_ANY = 0,  //!< Mark element if any channel is NaN
+    MASK_NANS = 1, //!< Make mask for NaN values
+    MASK_INFS = 2, //!< Make mask for Inf values
```

Comment:
Do we really need to mix NAN/INF processing in this single function?
Creation of GOD function doesn't look as a good practice.

Currently this is used in this way:

```
nanMask(normali, validNormalMask, MASK_NANS | MASK_INFS | MASK_INV);
```

Statement asks for finite numbers.

maskOfNaN() - `nan only`
maskOfInf() - `inf only`  - do we need this at all?
maskOfFinite() - `!(nan or inf)`

`maskOfFinite` => `isFiniteMask` as alternative naming